ql/src/java/org/apache/hadoop/hive/ql/plan/MergeJoinWork.java

```diff
@@ -66,6 +67,13 @@ public void replaceRoots(Map<Operator<?>, Operator<?>> replacementMap) {
 
   @Override
   public void configureJobConf(JobConf job) {
+    for(Operator rootOp : getAllRootOperators()) {
```

Comment:
I think, it can be extracted to base. In MapWork and ReduceWork , for example, you can do same, at least for FileSinkOperator - iterate over getAllRootOperators() and call configureJobConf

Reply:
Done